### ISSUE TITLE
Model with dynamic key (aka dictionary)

### ISSUE THREAD

**Issue by Nargonath** [szokodiakos/typegoose#198]
Here is the end result I'm looking for:
```javascript
const customer = {
  name: 'Henri',
  partners: {
    microsoft: {
      country: 'FR',
    },
    google: {
      country: 'EN',
    },
  },
};
```
I can represent it in Typescript but I don't find a way with Typegoose. It yells errors at me (see below). Here is my attempt:
```typescript
export class Partner extends Typegoose {
  @prop({ required: true })
  country: string;
}

export class PartnerMap extends Typegoose {
  @prop({ ref: Partner })
  [partnerId: string]: Ref<Partner>;
}

export class Customer extends Typegoose {
  @prop()
  name: string;

  @prop({ ref: PartnerMap })
  partners: Ref<PartnerMap>;
}

export default new Customer().getModelForClass(Customer);
```

Here is the error that I get in VSCode:
```
[ts] Property 'buildSchema' of type '(name: any, schemaOptions: any, sch?: any) => any' is not assignable to string index type 'Ref<Partner>'.
[ts] Property 'getModelForClass' of type '<T>(t: T, { existingMongoose, schemaOptions, existingConnection }?: GetModelForClassOptions) => Model<InstanceType<PartnerMap>> & PartnerMap & T' is not assignable to string index type 'Ref<Partner>'.
[ts] Property 'setModelForClass' of type '<T>(t: T, { existingMongoose, schemaOptions, existingConnection }?: GetModelForClassOptions) => Model<InstanceType<PartnerMap>> & PartnerMap & T' is not assignable to string index type 'Ref<Partner>'.
```

Am I missing something here? Is it a use case handled by `Typegoose` or am I just hitting an edge case not covered by `Typegoose`?

Thanks for your help.

**Comment by hasezoey** [comment]
your errors are just because its missing these functions, this is a mistake that could be fixed with (branch) `feature/4`, and i made a discussion in #356 
so no, you are not doing anything wrong, currently the only fix is using `Partial<T>` or `Exclude<T>`

**Comment by Nargonath** [comment]
Alright, thanks @hasezoey for your answer. In the end as I couldn't figure it out I went using plain `mongoose` but it might be useful to others. 👍

**Comment by hasezoey** [comment]
btw
original issue / title is about 
> Model with dynamic key (aka dictionary)

which is implemented with `@mapProp({ of })` in #342 

this can be closed

@Ben305
